File: typescript/packages/wallets/crossmint/src/plugins/faucet.plugin.ts
import type { CrossmintApiClient } from "@crossmint/common-sdk-base";
import { Chain, PluginBase } from "@goat-sdk/core";
import type { EVMWalletClient } from "@goat-sdk/wallet-evm";
import { isChainSupportedByFaucet } from "../chains";
import { CrossmintFaucetService } from "./faucet.service";

export class FaucetPlugin extends PluginBase<EVMWalletClient> {
    constructor(client: CrossmintApiClient) {
        super("faucet", [new CrossmintFaucetService(client)]);
    }

    supportsChain(chain: Chain) {
        if (chain.type !== "evm") {
            return false;
        }

        if (!chain.id) {
            return false;
        }

        return isChainSupportedByFaucet(chain.id);
    }
}

export function faucetPlugin(client: CrossmintApiClient) {
    return () => {
        return new FaucetPlugin(client);
    };
}

File: python/src/wallets/crossmint/goat_wallets/crossmint/faucet_plugin.py
from typing import Callable
from goat import Chain, PluginBase
from .faucet import CrossmintFaucetService
from .api_client import CrossmintWalletsAPI
from .chains import is_chain_supported_by_faucet

class FaucetPlugin(PluginBase):
    def __init__(self, client: CrossmintWalletsAPI):
        super().__init__("faucet", [CrossmintFaucetService(client)])

    def supports_chain(self, chain: Chain) -> bool:
        if chain["type"] != "evm":
            return False
        if not chain["id"]:
            return False
        return is_chain_supported_by_faucet(chain["id"])

def faucet_plugin(client: CrossmintWalletsAPI) -> Callable[[], FaucetPlugin]:
    return lambda: FaucetPlugin(client)